apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/chainid/BlockchainConfigUpdater.java

```diff
@@ -105,4 +122,67 @@ private HeightConfig getConfigAtHeight(int targetHeight) {
             .map(height -> new HeightConfig(blockchainProperties.get(height)))
             .orElse(null);
     }
+
+    public HeightConfig getCurrentConfig() {
+        return blockchainConfig.getCurrentConfig();
+    }
+
+    /**
+     * Return correct ShardingSetting by specified (trim) height. We find any shard settings with height
+     * associated with height.
+     *
+     * @param trimHeight target height (trim height usually)
+     * @return found ShardingSettings value OR Optional.Empty value if not found
+     */
+    public Optional<ShardingSettings> getShardingSettingsByTrimHeight(int trimHeight) {
+        if (this.chain == null) {
+            String error = "Can't get 'ShardingSettings', because Chain configuration is not initialized ! That's strange actually...";
+            log.warn(error);
+            return Optional.empty();
+        }
+        Map<Integer, BlockchainProperties> blockchainProperties = chain.getBlockchainProperties();
+        if (blockchainProperties == null) { // very small chance, but who knows...?
+            String error = String.format("Missing any 'BlockchainProperties' by trimHeight '%s' !", trimHeight);
+            log.warn(error);
+            return Optional.empty();
+        }
+        if (trimHeight < 0) { // no chance to look for negative height value in configs
+            String error = String.format("'trimHeight' is negative trimHeight '%s' !", trimHeight);
+            log.warn(error);
+            return Optional.empty();
+        }
+        if (shardingSettingsMap == null) {
+            // lazy initialization and caching data inside LinkedMap for correct ordering
+            shardingSettingsMap =
+                blockchainProperties
+                    .values().stream()
+            .collect(Collectors.toMap(BlockchainProperties::getHeight, BlockchainProperties::getShardingSettings,
+                (oldValue, newValue) -> oldValue, LinkedHashMap::new));
+        }
+        if (log.isTraceEnabled()) {
+            log.trace("enabledShardingSettingsMap = " + shardingSettingsMap.toString());
+        }
+        return shardingSettingsMap
```

Comment:
Yes, that was possible, but I didn't like memory allocations in the method. So I decided do not touch that code and created smaller caching Map for only sharding settings.

Reply:
What's wrong with 'memory allocations'? Could you clarify? Besides that, if it is possible to use existing code - you must use it to avoid code duplication